astroimtools/cutout_tools.py

```diff
@@ -16,11 +16,267 @@
+
+        >>> catalog = Table(data=[ids, ra, dec, cutout_width, cutout_height], \
+                            names=['id', 'ra', 'dec', 'cutout_width', 'cutout_height'])
+    To get a list of PrimaryHDU objects:
+        >>> cutouts = cutout_tool('h_udf_wfc_b_drz_img.fits', catalog)
+    To save to fits file, (returns a list of file names):
+        >>> cutouts = cutout_tool('h_udf_wfc_b_drz_img.fits', catalog, to_fits=True)
+    If the above catalog table is saved in an ECSV file with the proper units information:
+        >>> cutouts = cutout_tool('h_udf_wfc_b_drz_img.fits', 'catalog.ecsv')
+
+    Parameters
+    ----------
+    image : str or `HDUList` or `PrimaryHDU` or `ImageHDU` or `CompImageHDU`
+        Image to cut from. If string is provided, it is assumed to be a
+        fits file path.
+    catalog : str or `astropy.table.table.Table`
+        Catalog table defining the sources to cut out. Must contain
+        unit information as the cutouttool does not assume default units.
+    image_ext : int, optional
+        Image extension to extract header and data. Default is 0.
+    origin : int
+        Whether pixel coordinates are 0 or 1-basedpixel coordinates.
+    to_fits : bool
+        Save cutouts to fits files. Each cutout will be saved to a separate file.
+    output_dir : str
+        Path to directory to save the cutouts in. Set to
+        current_working_dir/cutouts by default
+    overwrite: bool, optional
+        Overwrite existing files. Default is `False`.
+    catalog_format : str, BaseReader
+        Input catalog format if reading from a file. Default is `ascii.ecsv`.
+        Note: Catalog format should be capable of staring units as metadata.
+    delimiter: str
+        Input catalog column delimiter string if reading from a file.
+    suppress_rotation : bool
+        Suppress rotation even if ``'cutout_pa'`` is provided. Default is `False`.
+    verbose : bool, optional
+        Print extra info. Default is `True`.
+
+    Returns
+    -------
+    cutouts : list
+        A list of PrimaryHDU or file names (if saving to fits).
+        If cutout failed for a target, `None` will be added as a
+        place holder.
+    """
+    # Optional dependencies...
+    from reproject.interpolation.high_level import reproject_interp
+
+    # read in the catalog file:
+    if isinstance(catalog, str):
+        if delimiter is None:
+            catalog = QTable.read(catalog, format=catalog_format)
+        else:
+            catalog = QTable.read(catalog, format=catalog_format, delimiter=delimiter)
+    elif not isinstance(catalog, Table):
+        raise TypeError("Catalog should be an astropy.table.table.Table or"
+                        " file name, got {0} instead".format(type(catalog)))
+
+    # Get data and wcs from ImageHDU:
+    if isinstance(image, str):
+        with fits.open(image) as pf:
+            image_hdu = pf[image_ext]
+            data = image_hdu.data
```

Comment:
Might want to check to make sure the data attribute exists, and actually contains data. As we saw with cubeviz, there are times people are sending in odd data or without data.